pkg/ipcache/ipcache.go

```diff
@@ -555,7 +564,12 @@ func (ipc *IPCache) DumpToListenerLocked(listener IPIdentityMappingListener) {
 		}
 		nodeID := uint16(0)
 		if hostIP != nil {
-			nodeID = ipc.AllocateNodeID(hostIP)
+			exists := false
+
+			nodeID, exists = ipc.GetNodeID(hostIP)
+			if !exists {
+				nodeID = ipc.AllocateNodeID(hostIP)
```

Comment:
My thought process on this is if we find an entry in the ipcache with a hostIP and no nodeID then we somehow got in a state where we have an entry that is missing a nodeID. So we should try to fix it by creating an ID for it.

The AllocateNodeID will lock the linuxNodeHandler and then rechecks if the nodeIP has a NodeID so I think we are OK from locking side because the AllocateNodeID would just return it if something allocated it between the check and call to Alocate.

What deletes it? Interesting question I guess it depends on how it got here in the first place. But a NodeDelete with the hostIP associated with it would then delete the entry. Or the podDelete for the ipcache entry would also do it.

I don't think this is essential to have so we can either remove it or leave it. I guess this logic is called from cmd on daemon side so really doesn't add much value and adds extra complexity we have to reason about. I'm tempted to just delete it.

Reply:
I replaced the Allocate with a Warning so we get some indication this is happening. I think this makes sense. wdyt